WeakAurasOptions/OptionsFrames/OptionsFrame.lua

```diff
@@ -297,134 +237,131 @@ function OptionsPrivate.CreateFrame()
       else
         self.codereview.frame:Hide()
       end
       if self.window == "newView" then
+        OptionsPrivate.SetTitle(L["New Template"])
         self.newView.frame:Show()
       else
         if self.newView then
           self.newView.frame:Hide()
         end
       end
       if self.window == "update" then
+        OptionsPrivate.SetTitle(L["Update"])
         self.update.frame:Show()
       else
         self.update.frame:Hide()
       end
       if self.window == "debuglog" then
+        OptionsPrivate.SetTitle(L["Debug Log"])
         self.debugLog.frame:Show()
       else
         self.debugLog.frame:Hide()
       end
       if self.window == "default" then
         if self.loadProgessVisible then
           self.loadProgress:Show()
-          self.toolbarContainer.frame:Hide()
+          self.toolbarContainer:Hide()
           self.filterInput:Hide();
         else
           self.loadProgress:Hide()
-          self.toolbarContainer.frame:Show()
+          self.toolbarContainer:Show()
           self.filterInput:Show();
           --self.filterInputClear:Show();
         end
       else
         self.loadProgress:Hide()
-        self.toolbarContainer.frame:Hide()
+        self.toolbarContainer:Hide()
         self.filterInput:Hide();
       end
     end
   end
 
-  local minimizebutton = CreateFrame("Button", nil, minimize)
-  minimizebutton:SetWidth(30)
-  minimizebutton:SetHeight(30)
-  minimizebutton:SetPoint("CENTER", minimize, "CENTER", 1, -1)
-  minimizebutton:SetNormalTexture("Interface\\BUTTONS\\UI-Panel-CollapseButton-Up.blp")
-  minimizebutton:SetPushedTexture("Interface\\BUTTONS\\UI-Panel-CollapseButton-Down.blp")
-  minimizebutton:SetHighlightTexture("Interface\\BUTTONS\\UI-Panel-MinimizeButton-Highlight.blp")
-  minimizebutton:SetScript("OnClick", function()
-    if frame.minimized then
-      frame.minimized = nil
-      if odb.frame then
-        if not odb.frame.height or odb.frame.height < 240 then
-          odb.frame.height = 500
-        end
-      end
-      frame:SetHeight(odb.frame and odb.frame.height or 500)
-      minimizebutton:SetNormalTexture("Interface\\BUTTONS\\UI-Panel-CollapseButton-Up.blp")
-      minimizebutton:SetPushedTexture("Interface\\BUTTONS\\UI-Panel-CollapseButton-Down.blp")
 
-      frame.buttonsScroll:DoLayout()
-    else
-      frame.minimized = true
-      frame:SetHeight(40)
-      minimizebutton:SetNormalTexture("Interface\\BUTTONS\\UI-Panel-ExpandButton-Up.blp")
-      minimizebutton:SetPushedTexture("Interface\\BUTTONS\\UI-Panel-ExpandButton-Down.blp")
-    end
+
+  local minimizebutton = CreateFrame("Button", nil, frame, "MaximizeMinimizeButtonFrameTemplate")
+  minimizebutton:SetPoint("RIGHT", frame.CloseButton, "LEFT", 0, 0)
+  minimizebutton:SetOnMaximizedCallback(function()
+    frame.minimized = false
+    frame:SetHeight(odb.frame and odb.frame.height or defaultHeight)
+    frame:SetWidth(odb.frame and odb.frame.width or defaultWidth)
```

Comment:
As discussed on Discord, I think it looks better with the previous behavior because then the frame does not jump around